projects/igniteui-angular/migrations/common/tsUtils.ts

```diff
@@ -314,16 +320,45 @@ export const getTypeDefinitionAtPosition =
         return null;
     };
 
+
+/**
+ * Determines if a member belongs to a type in the `igniteui-angular` toolkit.
+ *
+ * @param change The change that will be applied.
+ * @param langServ The Typescript/Angular Language Service
+ * @param entryPath Relative file path.
+ * @param matchPosition The position of the identifier.
+ */
 export const isMemberIgniteUI =
     (change: MemberChange, langServ: tss.LanguageService, entryPath: string, matchPosition: number): boolean => {
-        const prevChar = langServ.getProgram().getSourceFile(entryPath).getText().substr(matchPosition - 2, 1);
-        if (prevChar === ')') {
+        const content = langServ.getProgram().getSourceFile(entryPath).getText();
+        matchPosition = shiftMatchPosition(matchPosition, content);
+        const prevChar = content.substr(matchPosition - 1, 1);
+        if (prevChar === SpecialIdentifiers.ClosingParenthesis) {
             // methodCall().identifier
-            matchPosition = langServ.getBraceMatchingAtPosition(entryPath, matchPosition - 2)[0]?.start ?? matchPosition;
+            matchPosition = langServ.getBraceMatchingAtPosition(entryPath, matchPosition - 1)[0]?.start ?? matchPosition;
+        }
 
+        const typeDef = getTypeDefinitionAtPosition(langServ, entryPath, matchPosition);
+        if (!typeDef) {
+            return false;
         }
-        const typeDef = getTypeDefinitionAtPosition(langServ, entryPath, matchPosition - 1);
-        return !typeDef ? false : typeDef.fileName.includes(IG_PACKAGE_NAME) && change.definedIn.indexOf(typeDef.name) !== -1;
+
+        return typeDef.fileName.includes(IG_PACKAGE_NAME)
+            && (change.definedIn.indexOf(typeDef.name) !== -1
+                || change.definedIn.indexOf(typeDef.containerName) !== -1);
     };
 
+/**
+ * Shifts the match position of the identifier to the left
+ * until any character other than an empty string or a '.' is reached. #9347
+ */
+const shiftMatchPosition = (matchPosition: number, content: string): number => {
+    do {
+        matchPosition--;
+    }
+    while (!content[matchPosition - 1].trim() || content[matchPosition - 1] === SpecialIdentifiers.MemberAccess);
```

Comment:
Also also, really weird with the do-while, since you are essentially decrementing _and_ looking at the prev char at the same time, basically 2 positions back, no? I know you mean to replace the old match - 1 position, but this doesn't make for a good general purpose method :)
Bonus: Would this not fail for a scenario like `e.member` since it'll jump over the dot immediately and there's nothing in front the `e`?